### PR TITLE
Export code cleanup - Use getComponentPaymentFields from processorClass

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Minor code refactor as part of ongoing cleanup on export processor

Before
----------------------------------------
Less functions on new processor class

After
----------------------------------------
More functions on new processor class

Technical Details
----------------------------------------
We have been cleaning up this class preliminary to address bugs, including moving code to use the new ExportProcessor class

Comments
----------------------------------------

@twomice I've added another small refactor - this affects the compilation of payment fields being exported (eg from a participant export or member export) - it seems they should be included in 'primary fields' for the entity or be selectable for export - but the way this functionality was added is pretty bad.

Next step is to move getTransformedFieldValue off to the ExportProcessor class & extracting a buildRow function. The goal would be to merge the household in that function (maybe via array merge) and completely eliminate the mergeSameHousehold function

### CODE REVIEW

File: CRM/Export/BAO/Export.php [hunk 1]
```diff
@@ -432,25 +432,16 @@ public static function exportComponents(
...
     $paymentDetails = array();
     if ($processor->isExportPaymentFields()) {
-
       // get payment related in for event and members
       $paymentDetails = CRM_Contribute_BAO_Contribution::getContributionDetails($exportMode, $ids);
       //get all payment headers.
       // If we haven't selected specific payment fields, load in all the
       // payment headers.
       if (!$processor->isExportSpecifiedPaymentFields()) {
-        $paymentHeaders = self::componentPaymentFields();
```
Comment (eileenmcnaughton) [inline CRM/Export/BAO/Export.php]: Replaced by $processor->getPaymentHeaders(

File: CRM/Export/BAO/Export.php [hunk 2]
```diff
@@ -432,25 +432,16 @@ public static function exportComponents(
...
           $addPaymentHeader = TRUE;
         }
       }
-      // If we have selected specific payment fields, leave the payment headers
-      // as an empty array; the headers for each selected field will be added
-      // elsewhere.
-      else {
-        $paymentHeaders = array();
-      }
-      $nullContributionDetails = array_fill_keys(array_keys($paymentHeaders), NULL);
```
Comment (eileenmcnaughton) [inline CRM/Export/BAO/Export.php]: calculated when it is used

File: CRM/Export/BAO/Export.php [hunk 3]
```diff
@@ -468,6 +459,16 @@ public static function exportComponents(
 
     list($outputColumns, $headerRows, $sqlColumns, $metadata) = self::getExportStructureArrays($returnProperties, $processor);
 
+    // add payment headers if required
+    if ($addPaymentHeader && $processor->isExportPaymentFields()) {
```
Comment (eileenmcnaughton) [inline CRM/Export/BAO/Export.php]: moved to being before the row-by-row loop

**Comment by twomice** [comment]
@eileenmcnaughton The code looks reasonable to me; I don't have any improvements to suggest. I also don't expect that manual testing would turn up anything useful, but can do it if you think it's helpful.

**Comment by eileenmcnaughton** [comment]
@twomice so in general we do try to test each change - I'm OK with merging this & then doing the next one if you commit to do a thorough round of testing towards the end of the month (ie. test all at once rather than individually)

**Comment by twomice** [comment]
@eileenmcnaughton If you're saying manual testing would be appropriate here, I can do that. I suppose it would be good to try a few different exports with primary fields, custom fields, memberships, and contributions?

**Comment by eileenmcnaughton** [comment]
@twomice In this case the focus would be on exporting entities with contributions - ie. some entities support (or at least according to the code :-) exporting related contribution data

**Comment by twomice** [comment]
Great, will do. Thanks!

**Comment by mattwire** [comment]
@eileenmcnaughton I've done some basic testing with this PR:
- Participant export, standard fields ~151contacts: Exports ok, export has contact, participant, contribution details where appropriate.
- Membership export, standard fields ~150contacts: Exports ok, export has contact, membership, contribution details where appropriate.
- Contact export, standard fields ~200contacts: Exports ok, export has contact details.  Merge same address works results in about 150 results, Merge households crashes as before (but this PR is not fixing that).

**Pull request merged**